### PR TITLE
[js] test order of host callout & typecheck during eval

### PR DESCRIPTION
**Pull request by mikesamuel**
This adds tests for tc39/ecma262#1504 which cannot be tested purely in
JavaScript because it involves HostEnsureCanCompileStrings.

As documented in a table on tc39/ecma262#1495 this passes on recent
Chrome and Safari and fails on Firefox so this test probably shouldn't
land right away.

### CODE REVIEW

**Comment by jugglinmike** [comment]
Mind moving this to the `content-security-policy/` directory?

**Comment by mikesamuel** [comment]
@jugglinmike I don't mind at all, but it's testing language in ECMA262, not language in CSP3.

**Comment by jugglinmike** [comment]
[It's kind of defined in both places.](https://w3c.github.io/webappsec-csp/#ecma-integration) Storing the test in `content-security-policy` more clearly relates the functionality to the web platform, and I'm thinking it makes it more visible for CSP's editors.

Alternatively, you might consider testing this in Test262 via some new host-defined behavior along [the same lines as `IsHTMLDDA`/`document.all`](https://github.com/tc39/test262/pull/1305). I haven't review the semantics, though, so it might not be a good match for this.

**Comment by mikesamuel** [comment]
@jugglinmike Will move it over.

**Comment by mikesamuel** [comment]
@jugglinmike Moved and reworked to follow conventions local to that directory.

File: content-security-policy/generic/eval-typecheck-callout-order.html
Comment (jugglinmike) [inline content-security-policy/generic/eval-typecheck-callout-order.html]: Did you mean for this to go in the `<head>`?

**Comment by jugglinmike** [comment]
> @jugglinmike Moved and reworked to follow conventions local to that directory.

Thanks!

> As documented in a table on tc39/ecma262#1495 this passes on recent Chrome and Safari and fails on Firefox so this test probably shouldn't land right away.

We can add "do not merge yet" to the pull request, or we could merge it with the ".tentative" file name marker. The latter is less likely to be forgotten, so if this isn't contentious, then that might be best. @mikewest do you have a preference?

File: content-security-policy/generic/eval-typecheck-callout-order.html
Comment (mikesamuel) [reply content-security-policy/generic/eval-typecheck-callout-order.html]: Yes.  Fixed.

**Comment by mikewest** [comment]
The test seems accurate to me, assuming https://github.com/tc39/ecma262/pull/1504 lands. I'd be fine with y'all landing these tests now as `.tentative`, or waiting until that patch lands in ecmascript and landing the patches without that designation. Either way is great!

**Comment by mikesamuel** [comment]
The test is now tentative.

```sh
$ git mv content-security-policy/generic/eval-typecheck-callout-order{,.tentative}.html
$ git mv content-security-policy/generic/eval-typecheck-callout-order{,.tentative}.html.headers
```

**Review by jugglinmike — APPROVED** [review]
Looks good to me

**Pull request merged**
